pkg/dataobj/internal/dataset/value.go

```diff
@@ -12,6 +13,7 @@ import (
 // Helper types
 type (
 	stringptr *byte
+	bytearray []byte
```

Comment:
I believe this will allocate when calling `ByteArrayValue` because the slice header gets wrapped in a pointer which escapes to the heap; we may want to define this as `bytearray *byte` and use `unsafe.SliceData`/`unsafe.Slice` to cast it back and forth.